sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -29,6 +29,45 @@ template <class To, class From> To cast(From value) {
   return (To)(value);
 }
 
+// USM helper function to get an extension function pointer
+template <typename T>
+pi_result getExtFuncFromContext(pi_context context, const char *func, T *fptr) {
+  size_t deviceCount;
+  cl_int ret_err = clGetContextInfo(
+      cast<cl_context>(context), CL_CONTEXT_DEVICES, 0, nullptr, &deviceCount);
+
+  std::vector<cl_device_id> devicesInCtx(deviceCount);
+
+  if (ret_err != CL_SUCCESS || deviceCount < 1) {
+    return cast<pi_result>(CL_INVALID_CONTEXT);
```

Comment:
Is it possible to return PI value directly instead of casting OpenCL macro? If so, please, apply to all instances.
```suggestion
    return PI_INVALID_CONTEXT;
```
or cast OpenCL return value:
```suggestion
    return cast<pi_result>(ret_err);
```

Reply:
Good catch.